pkg/sql/opt_catalog.go

```diff
@@ -766,18 +780,19 @@ func newOptTable(
 		} else {
 			ot.indexes[i].init(ot, i, idxDesc, idxZone, -1 /* virtualColOrd */)
 		}
-	}
 
-	ot.uniqueConstraints = make([]optUniqueConstraint, 0, len(ot.desc.UniqueWithoutIndexConstraints))
-	for i := range ot.desc.UniqueWithoutIndexConstraints {
-		u := &ot.desc.UniqueWithoutIndexConstraints[i]
-		ot.uniqueConstraints = append(ot.uniqueConstraints, optUniqueConstraint{
-			name:         u.Name,
-			table:        ot.ID(),
-			columns:      u.ColumnIDs,
-			withoutIndex: true,
-			validity:     u.Validity,
-		})
+		// Add unique constraints for implicitly partitioned unique indexes.
+		if idxDesc.Unique && idxDesc.Partitioning.NumImplicitColumns > 0 {
```

Comment:
do we want this to be supported for hash partitioned indexes as well?
if so, maybe it's worth using `ExplicitColumnStartIdx`

Reply:
oh nvm, looks like that is handled differnetly!